Client/ViewModels/MonthlyBudgetViewModel.razor.cs

```diff
@@ -19,27 +25,63 @@ public MonthlyBudgetViewModel(HttpClient apiClient, NavigationManager navManager
             monthlyBudgetService = mbService;
         }
 
-        public async Task GetLatestMonthlyBudgetForUser() {
+        public async Task SetupLatestMonthlyBudgetForUser() {
             
-            appState.monthlyBudgetModel = await monthlyBudgetService.GetLatestMonthlyBudget();
-            /*DateTime today = DateTime.Now;
-            int month = today.Month;
-            int year = today.Year;
-            appState.monthlyBudgetModel = await monthlyBudgetService.GetMonthlyBudget(month, year);
-            if(appState.monthlyBudgetModel == null)
-            {
-                appState.monthlyBudgetModel.Income = 0;
-                appState.monthlyBudgetModel.Outgoing = 0;
-                appState.monthlyBudgetModel.Remaining = 0;
-            }*/
+           await GetAllMonthlyBudgetsForUser();
+
+           GetAvailableYearsForUser();
+           selectedYear = years.LastOrDefault();
+
+           GetMonthsForSelectedYear();
+           selectedMonth = monthsForYear.LastOrDefault();
+
+           appState.monthlyBudgetModel = allMonthlyBudgetsForUser.OrderByDescending(x => x.Year)
+                    .ThenByDescending(x => x.Month)
+                    .FirstOrDefault();
         }
 
-        public async Task GetAvailableYearsForUser() {
+        public async Task ReloadBudgetForSelectedYear() {
 
+           GetMonthsForSelectedYear();
+           selectedMonth = monthsForYear.FirstOrDefault();
+           appState.monthlyBudgetModel = await monthlyBudgetService.GetMonthlyBudget(selectedMonth.MonthNumber, selectedYear);
+        }
+
+        public async Task GetAllMonthlyBudgetsForUser() {
+
+            allMonthlyBudgetsForUser = await monthlyBudgetService.GetAllMonthlyBudgetDatesForUser();
+            
         }
+        public void GetAvailableYearsForUser() {
+            
+            years = allMonthlyBudgetsForUser.OrderBy(x => x.Year)
+            .Select(x => x.Year)
+            .Distinct()
+            .ToList();
+        }
+
+        public async Task GetSpecificMonthlyBudget() {
+
+            appState.monthlyBudgetModel = await monthlyBudgetService.GetMonthlyBudget(selectedMonth.MonthNumber, selectedYear);
+        }
+
+        public void GetMonthsForSelectedYear() {
 
-        public async Task GetAvailableMonthsForSelectedYear() {
+            var months = allMonthlyBudgetsForUser.Where(x => x.Year == selectedYear)
+            .OrderBy(x => x.Month)
+            .Select(x => x.Month)
+            .Distinct()
+            .ToList();
 
+            monthsForYear.Clear();
+            months.ForEach(delegate(int month) {
+                monthsForYear.Add(
+                    new MonthModel {
+                        MonthName = CultureInfo.CurrentCulture.DateTimeFormat.GetMonthName(month),
+                        MonthNumber = month
+                    }
+                );           
+            });
```

Comment:
interesting implementation here 😜 

```suggestion
            monthsForYear = months.Select(month => 
                new MonthModel {
                    MonthName = CultureInfo.CurrentCulture.DateTimeFormat.GetMonthName(month),
                    MonthNumber = month
                }         
            );
```

This _should_ work...

Reply:
I made the change kind of as described. I was also able to remove the rest of the code from the function to one statement